### PR TITLE
CIV-5195 Full Reject Fields to CCD

### PR DESCRIPTION
**Pull request by hmcts-version1-domini**
### JIRA link ###

https://tools.hmcts.net/jira/browse/CIV-5195

### Change description ###

Develop the mapping between CUI response data and CCD, for 'Reject - I've paid what I believe I owe' and 'Reject - I dispute all of the claim' and 'Reject - I dispute the claim and want to make a counterclaim' response types.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by kenneth-hmcts — DISMISSED** [review]

File: src/main/services/translation/response/ccdTranslation.ts [hunk 1]
```diff
@@ -33,11 +35,14 @@ export const translateDraftResponseToCCD = (claim: Claim, addressHasChange: bool
...
     respondent1RepaymentPlan: toCCDRepaymentPlan(claim.partialAdmission?.paymentIntention?.repaymentPlan),
-    respondToClaimAdmitPartLRspec: toCCDPayBySetDate(claim.partialAdmission.paymentIntention.paymentDate),
+    respondToClaimAdmitPartLRspec: toCCDPayBySetDate(claim.partialAdmission?.paymentIntention?.paymentDate),
     responseClaimMediationSpecRequired: toAgreedMediation(claim.mediation),
     specAoSApplicantCorrespondenceAddressRequired: addressHasChange ? YesNoUpperCamelCase.NO : YesNoUpperCamelCase.YES,
     totalClaimAmount: claim.totalClaimAmount,
     respondent1: toCCDParty(claim.respondent1),
+    defenceRouteRequired: toCCDRejectAllOfClaimType(claim.rejectAllOfClaim?.option),
+    respondToClaim: toCCDRespondToClaim(claim.rejectAllOfClaim?.howMuchHaveYouPaid),
+    detailsOfWhyDoesYouDisputeTheClaim: claim.rejectAllOfClaim?.defence?.text ? claim.rejectAllOfClaim?.defence?.text : claim.rejectAllOfClaim?.whyDoYouDisagree?.text,
```
Comment (laperla11) [inline src/main/services/translation/response/ccdTranslation.ts:45]: you can use this short version:
`detailsOfWhyDoesYouDisputeTheClaim: claim.rejectAllOfClaim?.defence?.text ??claim.rejectAllOfClaim?.whyDoYouDisagree?.text,`

File: src/test/unit/services/translation/response/convertToCCDRejectAllOfClaimType.test.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import {Claim} from 'models/claim';
+import {RejectAllOfClaimType} from 'form/models/rejectAllOfClaimType';
+import {CCDRejectAllOfClaimType} from 'models/ccdResponse/ccdRejectAllOfClaimType';
+import {RejectAllOfClaim} from 'form/models/rejectAllOfClaim';
+import {toCCDRejectAllOfClaimType} from 'services/translation/response/convertToCCDRejectAllOfClaimType';
+
+describe('translate RejectAllOfClaimType to CCD model', () => {
```
Comment (laperla11) [inline src/test/unit/services/translation/response/convertToCCDRejectAllOfClaimType.test.ts:7]: Pls use Given, When , Then pattern in tests

**Review by laperla11 — CHANGES_REQUESTED** [review]

File: src/main/services/translation/response/ccdTranslation.ts (hunk 1 above)
Comment (hmcts-version1-domini) [reply src/main/services/translation/response/ccdTranslation.ts]: Good! Done. Thanks!

File: src/test/unit/services/translation/response/convertToCCDRejectAllOfClaimType.test.ts (hunk 2 above)
Comment (hmcts-version1-domini) [reply src/test/unit/services/translation/response/convertToCCDRejectAllOfClaimType.test.ts:7]: Right! Done. Thanks!

**Review by kenneth-hmcts — DISMISSED** [review]

**Review by laperla11 — APPROVED** [review]

**Review by dtortolaV1 — DISMISSED** [review]

**Review by portizdev — APPROVED** [review]

**Review by hmcts-version1-ignacio — APPROVED** [review]

**Pull request merged**